config/checkstyle-non-main-files-suppressions.xml

```diff
@@ -60,6 +60,8 @@
   <!-- test resources are weird by design, no validations in them -->
   <suppress id="lineLength"
          files="src[\\/]it[\\/]resources[\\/].*[\\/]InputLineLength.*\.java"/>
+  <suppress id="lineLength"
+         files="src[\\/]it[\\/]resources[\\/].*[\\/]InputPackageStatement.*\.java"/>
```

Comment:
Can we use line numbers of suppression?
We pretty confident that package will be at top of file.
But no other lines should violate such rule.
https://checkstyle.org/filters/suppressionfilter.html#Examples
There's `lines` attribute.

Reply:
There are 2 `InputPackageStatement.java` files, one inside `toolongpackagetotestcoveragegooglesjavastylerule` package and other inside `rule32packagestatement` package. 

The suppression you're suggesting would be only valid for the input file inside `toolongpackagetotestcoveragegooglesjavastylerule`. We would still need to use the existing suppression for other input file.

Reply:
add extra property/attribute - `lines` to suppression to minimize area of suppression.

Reply:
there is even better idea , please use 
and `// violation 'package statement should not be line-wrapped.'`

Reply:
done

https://github.com/checkstyle/checkstyle/blob/6521bff80ec0ca170e7f614534bf7d7cf8c39e35/config/checkstyle-non-main-files-suppressions.xml#L74-L76

Reply:
See solution below